src/transaction/log_page_buffer.c

```diff
@@ -1559,6 +1565,103 @@ logpb_fetch_header_from_active_log (THREAD_ENTRY * thread_p, const char *db_full
   return error_code;
 }
 
+// it peeks header page of the backuped log active file
+static int
+logpb_peek_header_of_active_log_from_backup (THREAD_ENTRY * thread_p, const char *active_log_path, LOG_HEADER * hdr)
+{
+  char log_pgbuf[IO_MAX_PAGE_SIZE + MAX_ALIGNMENT], *aligned_log_pgbuf;
+  LOG_HEADER *log_hdr;
+  LOG_PHY_PAGEID phy_pageid;
+  int error_code = NO_ERROR;
+  LOG_PAGE *log_pgptr;
+
+  assert (active_log_path != NULL);
+  assert (hdr != NULL);
+
+  aligned_log_pgbuf = PTR_ALIGN (log_pgbuf, MAX_ALIGNMENT);
+
+  if (fileio_is_volume_exist (active_log_path) == false)
+    {
+      return ER_FAILED;
+    }
+
+  int log_vdes = fileio_open (active_log_path, O_RDONLY, 0);
+  if (log_vdes == NULL_VOLDES)
+    {
+      return ER_FAILED;
+    }
+
+  phy_pageid = LOGPB_PHYSICAL_HEADER_PAGE_ID;
+  logpb_log ("reading from active log:%s, physical page is : %lld\n", active_log_path, (long long int) phy_pageid);
+
+  log_pgptr = (LOG_PAGE *) aligned_log_pgbuf;
+
+  if (fileio_read (thread_p, log_vdes, log_pgptr, phy_pageid, LOG_PAGESIZE) == NULL)
+    {
+      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_LOG_READ, 3, LOGPB_HEADER_PAGE_ID, phy_pageid, active_log_path);
+      error_code = ER_LOG_READ;
+      goto end;
+    }
+
+  log_hdr = (LOG_HEADER *) (log_pgptr->area);
+  *hdr = *log_hdr;
+
+  if (log_pgptr->hdr.logical_pageid != LOGPB_HEADER_PAGE_ID || log_pgptr->hdr.offset != NULL_OFFSET)
+    {
+      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_LOG_PAGE_CORRUPTED, 1, phy_pageid);
+      error_code = ER_LOG_PAGE_CORRUPTED;
+      goto end;
+    }
+
+end:
+  fileio_close (log_vdes);
+
+  return error_code;
+}
+
+// it probes whether the next archive file exists
+// if the case, the backuped log active file is regarded as unuseful and restore will reset log.
+static bool
+logpb_is_log_active_from_backup_useful (THREAD_ENTRY * thread_p, const char *active_log_path, const char *db_full_name)
+{
+  LOG_HEADER hdr;
+
+  if (logpb_peek_header_of_active_log_from_backup (thread_p, active_log_path, &hdr) != NO_ERROR)
+    {
+      // something bad happened
+      return false;
+    }
+
+  // make next archive name
+  char next_archive_file_path[PATH_MAX], next_archive_file_name[PATH_MAX];
+  char *p;
+
+  strcpy (next_archive_file_path, active_log_path);
```

Comment:
can fileio_make_log_archive_name be used?

Reply:
fixed :)